test/functional/api/window_spec.lua

```diff
@@ -347,6 +347,21 @@ describe('API/win', function()
       eq(2, #meths.list_wins())
       eq('', funcs.getcmdwintype())
     end)
+
+    it('closing current (float) window of another tabpage #15313', function()
+      command('tabedit')
+      eq(2, eval('tabpagenr()'))
+      local win = meths.open_win(0, true, {
+        relative='editor', row=10, col=10, width=50, height=10
+      })
+      local tabpage = eval('tabpagenr()')
+      command('tabprevious')
+      eq(1, eval('tabpagenr()'))
+      meths.win_close(win, false)
+
+      eq(1000, meths.tabpage_get_win(tabpage))
```

Comment:
this is failing, and the linux CI failures look like maybe a crash?

Reply:
Thanks!
I found failure from `linux asan (cc=clang-12)`'s `View raw logs` and fixed it.
But I'm not sure why `.id` is needed.

https://pipelines.actions.githubusercontent.com/OAGhSzOoyagEUMnWXOouBvclYp4jdxeDQMweYOrPWiuB3OOcSA/_apis/pipelines/1/runs/7579/signedlogcontent/5?urlExpires=2021-08-22T21%3A51%3A33.8371685Z&urlSigningMethod=HMACV1&urlSignature=uIAyclUybrHzLFZIMkxjsPEVmisdNxq%2BkMMgFG6BQKk%3D
```
2021-08-22T16:22:17.2166679Z [1m[31m[  ERROR   ][0m[0m [36mtest/functional/api/window_spec.lua[0m @ [36m351[0m: [1mAPI/win close closing current (float) window of another tabpage #15313[0m
2021-08-22T16:22:17.2167520Z test/helpers.lua:73: Expected objects to be the same.
2021-08-22T16:22:17.2167983Z Passed in:
2021-08-22T16:22:17.2168350Z (table: 0x7f1f19d414b8) {
2021-08-22T16:22:17.2168694Z   [id] = 1001 }
2021-08-22T16:22:17.2169004Z Expected:
2021-08-22T16:22:17.2169336Z (number) 1000
```

Reply:
I think that error is indicating that the test framework thinks nvim_tabpage_get_win() is returning a list-like table.

    eq({ 1001 }, meths.tabpage_get_win(tabpage))